extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobNameTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Marco Primo
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DestinationBlobNameTest {
+    DestinationBlobName destinationBlobName = new DestinationBlobName();
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName,  null, 1, blobName",
+            "partName, blobName,  null, 2, partName",
+            "partName, blobName,  '', 1, blobName",
+            "partName, blobName,  '', 2, partName",
+    }, nullValues = { "null" })
+    void shouldReturnBlobName_whenFolderNameIsNullOrEmpty(String partName, String blobName, String folderName, int partSize, String expected) {
+        destinationBlobName.setBlobName(blobName);
+        destinationBlobName.setFolderName(folderName);
+        assertEquals(expected, destinationBlobName.resolve(partName, partSize));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, null, folderName, 1, folderName/partName",
+            "partName, null, folderName, 2, folderName/partName",
+            "partName, ''  , folderName, 1,  folderName/partName",
+            "partName, ''  , folderName, 2,  folderName/partName",
+            "partName, null, folderName/, 1, folderName/partName",
+            "partName, null, folderName/, 2, folderName/partName",
+            "partName, ''  , folderName/, 1, folderName/partName",
+            "partName, ''  , folderName/, 2, folderName/partName",
```

Comment:
no need to set different partSize here because as demonstrated by the cases there's no difference by having partSize equal 1 or 2